### ISSUE TITLE
Somfy RTS Sunscreen Open Close State

### ISSUE THREAD

**Issue by Veldkornet** [swttt/com.swttt.homekit#139]
HomeyKit v3.0.6
Homey v2.0.2

Would it be possible to switch the “Opened” and “Closed” status of the Somfy RTS Zonwering if the type is a sunscreen?

At the moment, when my sunscreen is closed (ie, it is “up” / not extended), it shows in HomeKit as “Open”. The reverse is obviously also true, when my sunscreen is extended (ie, open / down), it shows in HomeKit is “Closed”. 

![d509a7fe-df02-44b4-9a9c-3510a4604ec6](https://user-images.githubusercontent.com/10297361/52448778-41e07180-2b35-11e9-9445-152f3ee93459.jpeg)

Of course, if these were window blinds / curtains that would make sense. But for a sunscreen, it’s actually the opposite. 

Now I see in Homey, that you can choose what type of device the Somfy is. Would it be possible to reverse the “Open” / “Closed” status if the type is a sunscreen?

![f4823dcd-36ec-468a-8695-919b445e15c6](https://user-images.githubusercontent.com/10297361/52448825-663c4e00-2b35-11e9-8eb4-f62d707d8f0a.jpeg)

**Comment by robertklep** [comment]
At the moment, sun shades aren't supported at all (it's a new class for firmware v2 that we haven't yet implemented). Should be easy to add, including the inverted action.

**Comment by nhassink** [comment]
It is also possible to invert somfy sunscreens in the Homey app. This way iT works as expected in HomeKit (have them myself). IT is in the advanced settings menu

**Comment by robertklep** [comment]
@nhassink oh, that's even better! (less work for us)

Still need to add support for `sunshade` classes though (and possibly others too).

**Comment by Veldkornet** [comment]
But in Homey, it’s correct. Up is up and down is down, why would I want to break that one to fix HomeKit. 

Assume you’re referring to this, where you can invert the signal
![20e83d0b-7858-425b-a778-cdcf678db521](https://user-images.githubusercontent.com/10297361/52465461-fe125a00-2b7e-11e9-9599-a4d4c058015e.jpeg)

**Comment by nhassink** [comment]
Yes I have the second one set to ja.

**Comment by nhassink** [comment]
Another option is to use a virtual device and four flows to set the up/down state inverse to the actual sunscreen and add the virtual device to homekit. I had it this way when the inverse setting was not yet available in the HomeKit app

**Comment by Veldkornet** [comment]
Yes, I’ve done that previously with the virtual devices, but then I have two floating around in Homey, and I have to hide the one away somewhere so people don’t get confused etc. It’s messy.  But I don’t want workarounds for everything :) I just want it to work properly :)

**Comment by Andre92NL** [comment]
Hi, I am having the same issue with the inverted indication of the sunscreen and blinds (Somfy via TaHoma switch). Did you find a solution in the mean time? Thanks
